package-drone/src/main/java/de/dentrassi/pm/jenkins/DroneRecorder.java

```diff
@@ -485,19 +433,40 @@ public Void invoke ( final File basedir, final VirtualChannel channel ) throws I
                             filename = f;
                         }
                         uploader.upload ( file, filename );
-//                        TODO this.artifacts.put ( artifacId, f );
                     }
-
+                    uploadResult.addArtifacts ( uploader.complete () );
```

Comment:
Ok, it will. Than what is the the scope of complete method? If I want implement a new Uploader, what I should perform in the complete method? What does it returns now?